app/Services/Strava/DSWAnalysis/StravaActivityDswAnalysisScoringService.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace App\Services\Strava\DSWAnalysis;
+
+use App\Models\StravaActivity;
+use App\Models\StravaConnection;
+use App\Services\Strava\StravaActivityDswAnalysisScoreBandEnum;
+use Cache;
+use DB;
+use Illuminate\Support\Collection;
+
+class StravaActivityDswAnalysisScoringService
+{
+    public function getActivityScoreBand(StravaActivity $activity): StravaActivityDswAnalysisScoreBandEnum
+    {
+        if (is_null($activity->dswAnalysis)) {
+            return StravaActivityDswAnalysisScoreBandEnum::MISSING_ANALYSIS;
+        }
+
+        if (is_null($activity->average_heartrate)) {
+            return StravaActivityDswAnalysisScoreBandEnum::MISSING_HEARTRATE;
+        }
+
+        if (is_null($activity->average_watts)) {
+            return StravaActivityDswAnalysisScoreBandEnum::MISSING_POWER;
+        }
+
+        $scoreBands = $this->getScoreBandsByType($activity->rawActivity->stravaConnection);
+
+        $integerIntervals = (int) $activity->dswAnalysis->intervals;
+        $scoreKey = "{$activity->dswAnalysis->dsw_type_id}:{$integerIntervals}";
+
+        if (is_null($scoreBand = $scoreBands->get($scoreKey))) {
+            return StravaActivityDswAnalysisScoreBandEnum::NOT_ENOUGH_DATA;
+        }
+
+        if ($activity->dswAnalysis->dsw_score <= $scoreBand['band_1']) {
+            return StravaActivityDswAnalysisScoreBandEnum::BAND_1;
+        }
+
+        if ($activity->dswAnalysis->dsw_score <= $scoreBand['band_2']) {
+            return StravaActivityDswAnalysisScoreBandEnum::BAND_2;
+        }
+
+        return StravaActivityDswAnalysisScoreBandEnum::BAND_3;
+    }
+
+    public function getScoreBandsByType(StravaConnection $stravaConnection): Collection
+    {
+        $lastUpdatedDswAnalysis = StravaActivity::byUser($stravaConnection->user)
+            ->select(DB::raw('UNIX_TIMESTAMP(MAX(strava_activity_dsw_analyses.updated_at)) AS max_last_updated'))
+            ->join('strava_activity_dsw_analyses', 'strava_activities.id', '=', 'strava_activity_dsw_analyses.strava_activity_id')
+            ->first()?->max_last_updated ?? 'NULL';
+
+        return Cache::remember(
```

Comment:
This needs to be reworked so that the score band data is permanently saved to a cache key for the Strava connection, which is purged each time a DSW analysis record is created or updated.